src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -234,61 +237,55 @@ private boolean isTextSeparatedCorrectlyFromPrevious(String line, int columnNo)
+    private static boolean isSingleSpace(int[] line, int columnNo) {
+        return isSpace(line, columnNo) && !CommonUtil.isCodePointWhitespace(line, columnNo - 1);
     }
 
     /**
      * Checks if the {@code line} at {@code columnNo} is a space.
      *
-     * @param line The line in the file to examine.
+     * @param line unicode code point array of line in the file to examine.
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the character at {@code columnNo} is a space.
      */
-    private static boolean isSpace(String line, int columnNo) {
-        return line.charAt(columnNo) == ' ';
-    }
-
-    /**
-     * Checks if the {@code line} at {@code columnNo} is a whitespace character.
-     *
-     * @param line The line in the file to examine.
-     * @param columnNo The column position in the {@code line} to examine.
-     * @return {@code true} if the character at {@code columnNo} is a
-     *         whitespace.
-     */
-    private static boolean isWhitespace(String line, int columnNo) {
-        return Character.isWhitespace(line.charAt(columnNo));
+    private static boolean isSpace(int[] line, int columnNo) {
+        return line[columnNo] == ' ';
     }
 
     /**
      * Checks if the {@code line} up to and including {@code columnNo} is all
      * non-whitespace text encountered.
      *
-     * @param line The line in the file to examine.
+     * @param line unicode code point array of line in the file to examine.
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the column position is the first non-whitespace
      *         text on the {@code line}.
      */
-    private static boolean isFirstInLine(String line, int columnNo) {
-        return CommonUtil.isBlank(line.substring(0, columnNo));
+    private static boolean isFirstInLine(int[] line, int columnNo) {
+        return CodePointUtil.isBlank(Arrays.copyOfRange(line, 0, columnNo));
     }
 
     /**
      * Checks if the {@code line} at {@code columnNo} is the end of a comment,
      * '*&#47;'.
      *
-     * @param line The line in the file to examine.
+     * @param line unicode code point array of line in the file to examine.
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the previous text is a end comment block.
      */
-    private static boolean isBlockCommentEnd(String line, int columnNo) {
-        return line.substring(0, columnNo).trim().endsWith("*/");
+    private static boolean isBlockCommentEnd(int[] line, int columnNo) {
+        final int[] lineFromWhitespace = CodePointUtil.trim(Arrays.copyOfRange(line, 0, columnNo));
+        boolean isBlockCommentEnd = false;
+        if (lineFromWhitespace.length > 1) {
+            isBlockCommentEnd = CodePointUtil.startsWith(lineFromWhitespace, "*/",
```

Comment:
Why not create a method like `CodePointUtil.endsWith`?

Reply:
Because, endsWith can be achieved by changing `fromIndex` param. (somehow endsWith can also be used as startsWith, but startsWith sound more correct isn't it?)


Also,
`SingleSpaceSeparator` requires removing of trailing whitespace and checking for prefix from end.
`EmptyLineSeparator` requires removal of leading whitespace and check for prefix from end.
`SeparatorWrapCheck` requires removal of whitespace and checking if line becomes empty.

Should I create 3 separate function in CodePointUtil.
OR
Create a helper function in respective check classes.

Reply:
I would like to have all required methods that match the methods of the String class. I.e. `startsWith`, `endsWith`, `isBlank`, 'strip` and so on.

Reply:
Added `endsWith` and `stripTrailing` other required like stripLeading, startsWith is done in diff PR

Reply:
Missed this comment https://github.com/checkstyle/checkstyle/issues/10924#issuecomment-1028534612

So finally what should be the approach @pbludov @nmancus1 ?